x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/write_model/rule_execution_event.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RuleExecutionEventLevel } from '../common_model';
+
+// -----------------------------------------------------------------------------
+// Types of rule execution events in the write model
+
+export const RuleExecutionEventType = {
+  GENERIC: 'generic',
+  STATUS_CHANGED: 'status-changed',
+} as const;
```

Comment:
Yeah yeah, exactly, I share the same kind of vision. We definitely will be able to add more `RuleExecutionEventType`s if we need so. Doesn't mean we will have to. Cases where it would make sense to do later, imho:

- Extract existing stuff like 'gap' to separate `RuleExecutionEvent`s _if_ we want to log it earlier, i.e. exactly when they are captured/created in the code. And decouple them from status updates. Meaning that if we found a gap, we could log it as a separate event with a warning level, but delay or omit changing the status.
- New types of events with their own payload.

As for breaking `StatusChangedEvent` down to separate `StatusGoingToRunEvent`, `StatusSucceededEvent` - not sure, it would make sense only if they would have completely separate payloads and wouldn't have anything in common (in this case, it's a common status itself and its severity). Imho for now we could have a few factory methods for creating status events, but keep them as a single type.